util/formula_templater/testdata/boundary-desktop-cask.rb

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+cask "hashicorp-boundary-desktop" do
+  version "1.6.0"
+  sha256 "b6b5b15dfb469b7fdab9216788f5931e96561104de1ff7f9e0d6fda54701be09"
```

Comment:
I'm wondering, what's the homebrew behavior on arm64 if this is qualified by arch?
Say if instead we specified:

```
sha256 intel: "deadbeef"
```

Does it fail to install or does it fall back to amd64? (Might simplify our templating logic)

Reply:
I've been looking but can't find any documentation that describes the behavior. My assumption is it would fail, but given rosetta as an option, it may not :shrug:. I'll see about testing it locally to get a more definitive answer.

Reply:
It does not fallback:

> Error: Cask 'hashicorp-boundary-desktop' definition is invalid: invalid 'sha256' value: nil